WooCommerce/UITestsFoundation/Screens/MyStore/MyStoreScreen.swift

```diff
@@ -46,8 +47,11 @@ public final class MyStoreScreen: ScreenObject {
     }
 
     func verifyStatsForTimeframeLoaded(timeframe: String) -> MyStoreScreen {
-        let textPredicate = NSPredicate(format: "label MATCHES %@", "\(timeframe)")
-        XCTAssertTrue(app.staticTexts.containing(textPredicate).element.exists, "\(timeframe) chart not displayed")
+        let textPredicate = NSPredicate(format: "label MATCHES %@", timeframe)
+        let element = app.staticTexts.containing(textPredicate).element
+        let exists = element.waitForExistence(timeout: 30)
+
+        XCTAssertTrue(exists, "\(timeframe) chart not displayed")
```

Comment:
WDYT about renaming this variable for better readability?

```suggestion
        let elementExists = element.waitForExistence(timeout: 30)

        XCTAssertTrue(elementExists, "\(timeframe) chart not displayed")
```